pkgs/tools/compression/bzip2/default.nix

```diff
@@ -10,37 +9,41 @@
 
 stdenv.mkDerivation rec {
   pname = "bzip2";
-  version = "1.0.6.0.1";
+  version = "1.0.8";
 
-  /* We use versions patched to use autotools style properly,
-      saving lots of trouble. */
   src = fetchurl {
-    urls = map
-      (prefix: prefix + "/people/sbrabec/bzip2/tarballs/${pname}-${version}.tar.gz")
-      [
-        "http://ftp.uni-kl.de/pub/linux/suse"
-        "ftp://ftp.hs.uni-hamburg.de/pub/mirrors/suse"
-        "ftp://ftp.mplayerhq.hu/pub/linux/suse"
-        "http://ftp.suse.com/pub" # the original patched version but slow
-      ];
-    sha256 = "0b5b5p8c7bslc6fslcr1nj9136412v3qcvbg6yxi9argq9g72v8c";
+    url = "https://sourceware.org/pub/bzip2/${pname}-${version}.tar.gz";
+    sha256 = "0s92986cv0p692icqlw1j42y9nld8zd83qwhzbqd61p1dqbh6nmb";
   };
 
-  nativeBuildInputs = [ autoreconfHook ];
+  outputs = [ "bin" "dev" "out" "man" ] ++ lib.optionals linkStatic [ "static" ];
 
-  patches = [
-    ./CVE-2016-3189.patch
-    ./cve-2019-12900.patch
-  ];
-
-  postPatch = ''
-    sed -i -e '/<sys\\stat\.h>/s|\\|/|' bzip2.c
+  # fix shared library options
+  postPatch = stdenv.lib.optionalString stdenv.hostPlatform.isDarwin ''
+    substituteInPlace Makefile-libbz2_so \
+        --replace "soname" "install_name"
   '';
 
-  outputs = [ "bin" "dev" "out" "man" ];
+  buildPhase = ''
+    make CC=cc
+    make CC=cc -f Makefile-libbz2_so
+  '';
 
-  configureFlags =
-    stdenv.lib.optionals linkStatic [ "--enable-static" "--disable-shared" ];
+  installFlags = [ "PREFIX=$(out)" ];
+
+  postInstall = let
+      ext = stdenv.hostPlatform.extensions.sharedLibrary;
+    in ''
+    moveToOutput bin $bin
+    ln -s libbz2${ext}.1.0 libbz2${ext}.1
```

Comment:
The Makefile-libbz2_so build doesn't create a dylib so this shouldn't be conditional unless that's also changed.

Reply:
this used to be the case, but either works fine

```
-r-xr-xr-x 1 root nixbld 77168 Jan  1  1970 libbz2.1.dylib
lrwxr-xr-x 1 root nixbld    14 Jan  1  1970 libbz2.dylib -> libbz2.1.dylib
-r-xr-xr-x 1 root nixbld   944 Jan  1  1970 libbz2.la
```